specs/HubProtocol.md

```diff
@@ -348,8 +363,160 @@ World;0:;
+
+```
+0x95 0x01 0xa3 0x78 0x79 0x7a 0xc3 0xa6 0x6d 0x65 0x74 0x68 0x6f 0x64 0x91 0x2a
+```
+
+is decoded as follows:
+
+* `0x95` - 5-element array
+* `0x01` - `1` (Message Type - `Invocation` message)
+* `0xa3` - string of length 3 (Target)
+* `0x78` - `x`
+* `0x79` - `y`
+* `0x7a` - `z`
+* `0xc3` - `true` (NonBlocking)
+* `0xa6` - string of length 6 (Target)
+* `0x6d` - `m`
+* `0x65` - `e`
+* `0x74` - `t`
+* `0x68` - `h`
+* `0x6f` - `o`
+* `0x64` - `d`
+* `0x91` - 1-element array (Arguments)
+* `0x2a` - `42` (Argument value)
+
+### StreamItem Message Encoding
+
+`StreamItem` messages have the following structure:
+
+[2, InvocationId, Item]
+
+* `2` - Message Type - `2` indicates this is a `StreamItem` message
+* InvocationId - A `String` encoding the Invocation ID for the message
+* Item - the value of the stream item
+
+Example:
+
+The following payload:
+```
+0x93 0x02 0xa3 0x78 0x79 0x7a 0x2a
+```
+
+is decoded as follows:
+
+* `0x93` - 3-element array
+* `0x02` - `2` (Message Type - `StreamItem` message)
+* `0xa3` - string of length 3 (Target)
+* `0x78` - `x`
+* `0x79` - `y`
+* `0x7a` - `z`
+* `0x2a` - `42` (Item)
+
+### Completion Message Encoding
+
+`Completion` messages have the following structure
+
+```
+[3, InvocationId, ResultKind, Result?]
+```
+
+* `3` - Message Type - `3` indicates this is a `Completion` message
+* InvocationId - A `String` encoding the Invocation ID for the message
+* ResultKind - A flag indicating the invocation result kind:
+    * `1` - Error result - Result contains a `String` with the error message
+    * `2` - Void result - Result contains the value returned by the server
+    * `3` - Non-Void result - Result is absent
+* Result - An optional item containing the result of invocation. Absent if the server did not return any value (void methods)
+
+Examples:
+
+#### Error Result:
+
+The following payload:
+```
+0x94 0x03 0xa3 0x78 0x79 0x7a 0x01 0xa5 0x45 0x72 0x72 0x6f 0x72
+```
+
+is decoded as follows:
+
+* `0x94` - 4-element array
+* `0x03` - `3` (Message Type - `Result` message)
+* `0xa3` - string of length 3 (Target)
+* `0x78` - `x`
+* `0x79` - `y`
+* `0x7a` - `z`
+* `0x01` - `1` (ResultKind - Error result)
+* `0xa5` - string of lenght 5
+* `0x45` - `E`
+* `0x72` - `r`
+* `0x72` - `r`
+* `0x6f` - `o`
+* `0x72` - `r`
+
+#### Void Result:
+
+The following payload:
+```
+0x93 0x03 0xa3 0x78 0x79 0x7a 0x02
+```
+
+is decoded as follows:
+
+* `0x93` - 3-element array
+* `0x03` - `3` (Message Type - `Result` message)
+* `0xa3` - string of length 3 (Target)
+* `0x78` - `x`
+* `0x79` - `y`
+* `0x7a` - `z`
+* `0x02` - `2` (ResultKind - Void result)
+
+#### Non-Void Result:
+
+The following payload:
+```
+0x94 0x03 0xa3 0x78 0x79 0x7a 0x03 0x2a
+```
+
+is decoded as follows:
+
+* `0x94` - 4-element array
+* `0x03` - `3` (Message Type - `Result` message)
+* `0xa3` - string of length 3 (Target)
+* `0x78` - `x`
+* `0x79` - `y`
+* `0x7a` - `z`
+* `0x03` - `3` (ResultKind - Non-Void result)
+* `0x2a` - `42` (Result)
```

Comment:
no - result can be a primitive value, an array or on object (it's a method invocation result). The result is encoded using MsgPack rules. Describing how MsgPack encodes values is out of scope for this document by I added a link at the top of this section (https://github.com/msgpack/msgpack/blob/master/spec.md#formats).